Samples/Common/SSHShell.cs

```diff
@@ -0,0 +1,223 @@
+            {
+                string path = fromPath;
+                if (isUserHomeBased)
+                {
+                    path = homeDirectory + path;
+                }
+
+                using (MemoryStream mstream = new MemoryStream()) {
+                    scpClient.Download(path + "/" + fileName, mstream);
+                    if (mstream.Position >= maxCount)
+                    {
+                        return -1;
+                    }
+                    mstream.Position = 0;
+                    return mstream.Read(destBuff, 0, maxCount);
+                }
+            }
+            else
+            {
+                return -1;
+            }
+        }
+
+        /**
+         * Creates a new file on the remote host using the input content.
+         *
+         * @param data the byte array content to be uploaded
+         * @param fileName the name of the file for which the content will be saved into
+         * @param toPath the path of the file for which the content will be saved into
+         * @param isUserHomeBased true if the path of the file is relative to the user's home directory
+         * @param filePerm file permissions to be set
+         */
+        public void Upload(byte[] data, string fileName, string toPath, bool isUserHomeBased)
+        {
+            if (scpClient != null && sshClient != null)
+            {
+                string path = toPath;
+                if (isUserHomeBased)
+                {
+                    path = homeDirectory + path;
+                }
+
+                // Create the directory on the remote host
+                using (var command = sshClient.CreateCommand("mkdir -p " + path))
+                {
+                    command.Execute();
+                }
+
+                // Create the file containing the uploaded data
+                // byte[] data = System.Text.Encoding.ASCII.GetBytes("");
+                using (var mstream = new MemoryStream(data))
+                {
+                    scpClient.Upload(mstream, path + "/" + fileName);
+                }
+            }
+            // In Java
+            //ChannelSftp channel = (ChannelSftp) this.session.openChannel("sftp");
+            //channel.connect();
+            //String absolutePath = isUserHomeBased ? channel.getHome() + "/" + toPath : toPath;
+
+            //String path = "";
+            //for (String dir : absolutePath.split("/")) {
+            //    path = path + "/" + dir;
+            //    try
+            //    {
+            //        channel.mkdir(path);
+            //    }
+            //    catch (Exception ee)
+            //    {
+            //    }
+            //}
+            //channel.cd(absolutePath);
+            //channel.put(from, fileName);
+            //if (filePerm != null) {
+            //    channel.chmod(Integer.parseInt(filePerm), absolutePath + "/" + fileName);
+            //}
+            //channel.disconnect();
+        }
+
+        ~SSHShell()
```

Comment:
>~SSHShell() [](start = 8, length = 11)

how do you call this method? seems like you need to implement IDisposable pattern.